lib/services/local/users.go

```diff
@@ -669,7 +719,10 @@ func (s *IdentityService) UpsertGlobalWebauthnSessionData(ctx context.Context, s
 		return trace.BadParameter("missing parameter sd")
 	}
 
-	// TODO(codingllama): Limit number of in-flight challenges.
+	if entries := sdLimiter.add(scope, 1); entries > GlobalSessionDataMaxEntries {
+		sdLimiter.add(scope, -1) // request denied, adjust accordingly
+		return trace.LimitExceeded("too many in-flight challenges")
```

Comment:
should we decrement in error cases below? if it's intended to not decrement in those cases, maybe add a comment.